pkg/repair/repair.go

```diff
@@ -0,0 +1,273 @@
+	FieldSelectors                  string `json:"field_selectors"`
+	LabelSelectors                  string `json:"label_selectors"`
+}
+
+// The pod reconciler struct. Contains state used to reconcile broken pods.
+type BrokenPodReconciler struct {
+	client  client.Interface
+	Filters *Filters
+	Options *Options
+}
+
+// Constructs a new BrokenPodReconciler struct.
+func NewBrokenPodReconciler(client client.Interface, filters *Filters, options *Options) BrokenPodReconciler {
+	return BrokenPodReconciler{
+		client:  client,
+		Filters: filters,
+		Options: options,
+	}
+}
+
+// Label all pods detected as broken by ListPods with a customizable label
+func (bpr BrokenPodReconciler) LabelBrokenPods() (err error) {
+	// Get a list of all broken pods
+	podList, err := bpr.ListBrokenPods()
+	if err != nil {
+		return
+	}
+
+	for _, pod := range podList.Items {
+		labels := pod.GetLabels()
+		if _, ok := labels[bpr.Options.PodLabelKey]; ok {
+			log.Infof("Pod %s/%s already has label with key %s, skipping", pod.Namespace, pod.Name, bpr.Options.PodLabelKey)
+			continue
+		}
+
+		if labels == nil {
+			labels = map[string]string{}
+		}
+
+		log.Infof("Labeling pod %s/%s with label %s=%s", pod.Namespace, pod.Name, bpr.Options.PodLabelKey, bpr.Options.PodLabelValue)
+		labels[bpr.Options.PodLabelKey] = bpr.Options.PodLabelValue
+		pod.SetLabels(labels)
+
+		if _, err = bpr.client.CoreV1().Pods(pod.Namespace).Update(&pod); err != nil {
+			return
+		}
+	}
+	return
+}
+
+func (bpr BrokenPodReconciler) CreateEventsForBrokenPods() error {
+	// Get a list of all broken pods
+	podList, err := bpr.ListBrokenPods()
+	if err != nil {
+		return err
+	}
+
+	eventList, err := bpr.ListEvents()
+	if err != nil {
+		return err
+	}
+
+	var loopErrors []error
+	for _, pod := range podList.Items {
+		if event, ok := eventList[pod.UID]; ok {
+			log.Infof("Updating existing event for broken pod: %s/%s", pod.Namespace, pod.Name)
+			event.LastTimestamp = metav1.Now()
+			event.Count++
+			_, err := bpr.client.CoreV1().Events(pod.Namespace).Update(&event)
+			if err != nil {
+				loopErrors = append(loopErrors, err)
+			}
+		} else {
+			log.Infof("Creating event for broken pod: %s/%s", pod.Namespace, pod.Name)
+			_, err := bpr.client.CoreV1().Events(pod.Namespace).Create(&v1.Event{
+				Type: "Warning",
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName:      pod.Name,
+					Namespace:         pod.Namespace,
+					Generation:        0,
+					CreationTimestamp: metav1.Now(),
+					ClusterName:       pod.ClusterName,
+					Labels: map[string]string{
+						"istio-cni-daemonset-event": "true",
+					},
+				},
+				InvolvedObject: v1.ObjectReference{
+					Kind:            "pod",
+					Namespace:       pod.Namespace,
+					Name:            pod.Name,
+					UID:             pod.UID,
+					APIVersion:      pod.APIVersion,
+					ResourceVersion: pod.ResourceVersion,
+				},
+				Reason:         "BrokenIstioCNI",
```

Comment:
I don't think "broken" is the right term. It's not informative to users.
I would suggest renaming this everywhere into "unitialized"